### PR TITLE
CP-2816 prepare for dev VPN

### PR DESCRIPTION
**Pull request by bru**
this PR has mainly two effects:
1. it simplifies the `remote` command by delegating the ssh setup to the `sshconfig` command (in other words, it assumes your ssh configuration is already setup with the right options)
2. it adds the `--vpn` and `--nat` cli options to `sshconfig`, to control whether the config file will include the `ProxyCommand` setting (needed to tunnel through the NAT instances) or not.

### CODE REVIEW

**Comment by Azrael808** [comment]
This looks good to me, I did wonder perhaps about adding a section for all SSH connections at the top of the ssh config:

```
Host *
ServerAliveInterval 30
```

Not having this in my own config can be crippling on connections where the ISP aggressively terminates TCP sessions that appear idle.

**Comment by lackac** [comment]
@Azrael808 those are good options, but I'll leave them for individuals to add. I have a lot of stuff in my `.ssh/config` that I wouldn't like this command to mess with, so I think it's best to focus on just the C+ related hosts in this one.

**Comment by Azrael808** [comment]
@lackac ah, apologies, I didn't realise this was editing only a portion of the SSH config. Your comment makes complete sense!

File: libexec/cplus-ami-build [hunk 1]
```diff
@@ -29,8 +29,8 @@ BUILD_NUM=$2
 BUILD_BRANCH="production"
 INSTANCE_PROFILE="CollectPlus-AppTier-staging1-IamEc2InstanceProfile-TC2EEHP6ZC7S"
 CHEF_VERSION="12.18.31"
-BASTION_HOST="50.17.233.88"
-BASTION_USER="rails"
+BASTION_HOST=""
+BASTION_USER=""
```
Comment (bru) [inline libexec/cplus-ami-build]: @Azrael808 lgtm, only question is: will the resulting command (i.e. `packer build -var "bastion_host=" ...`) work as expected?

Comment (Azrael808) [reply libexec/cplus-ami-build]: Yes, I was using this method to produce the Parameter Store enabled AMIs.

**Pull request merged**